### ISSUE TITLE
Mathtext.py glyph mapping fails for StixFonts (UnicodeFonts subclass)

### ISSUE THREAD

**Issue by settheory** [matplotlib/matplotlib#7939]
The 'stixsans' mapping fails to find the glyph when the `font_class` is not in mapping dictionary.

Set the `mathtext.fontset` to `'stixsans'`, then try to create text objects. Mine failed for the symbol `'m'`.

```python
    import matplotlib as mpl
    mpl.rcParams['mathtext.fontset'] = 'stixsans'
```

In StixFonts (line 883 of mathtext.py):

```python
    def _map_virtual_font(self, fontname, font_class, uniindex):
        # Handle these "fonts" that are actually embedded in
        # other fonts.
        
        mapping = stix_virtual_fonts.get(fontname)
        if (self._sans and mapping is None and
            fontname not in ('regular', 'default')):
            mapping = stix_virtual_fonts['sf']
            doing_sans_conversion = True
        else:
            doing_sans_conversion = False

        if mapping is not None:
            if isinstance(mapping, dict):
                mapping = mapping.get(font_class, 'rm')
```

Note that `mapping.get(font_class, 'rm')` sets `'rm'` as the default value if `font_class` is not in the mapping. This should be changed to use `'rm'` as the default key: 

```python
    try:
        mapping = mapping[font_class]
    except KeyError:
        mapping = mapping['rm']
```

- The error message (truncated) is below

```
  File "C:\Anaconda3\lib\site-packages\pyparsing.py", line 1072, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "C:\Anaconda3\lib\site-packages\pyparsing.py", line 2658, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "C:\Anaconda3\lib\site-packages\pyparsing.py", line 1137, in _parseCache
    value = self._parseNoCache( instring, loc, doActions, callPreParse )
  File "C:\Anaconda3\lib\site-packages\pyparsing.py", line 1098, in _parseNoCache
    tokens = fn( instring, tokensStart, retTokens )
  File "C:\Anaconda3\lib\site-packages\pyparsing.py", line 819, in wrapper
    ret = func(*args[limit[0]:])
  File "C:\Anaconda3\lib\site-packages\matplotlib\mathtext.py", line 2510, in symbol
    return [Hlist( [self._make_space(0.2),
  File "C:\Anaconda3\lib\site-packages\matplotlib\mathtext.py", line 2464, in _make_space
    state.font, rcParams['mathtext.default'], 'm', state.fontsize, state.dpi)
  File "C:\Anaconda3\lib\site-packages\matplotlib\mathtext.py", line 466, in get_metrics
    info = self._get_info(font, font_class, sym, fontsize, dpi)
  File "C:\Anaconda3\lib\site-packages\matplotlib\mathtext.py", line 601, in _get_info
    self._get_glyph(fontname, font_class, sym, fontsize)
  File "C:\Anaconda3\lib\site-packages\matplotlib\mathtext.py", line 830, in _get_glyph
    fontname, font_class, uniindex)
  File "C:\Anaconda3\lib\site-packages\matplotlib\mathtext.py", line 942, in _map_virtual_font
    if uniindex < range[0]:
TypeError: unorderable types: int() < str()

```

**Matplotlib version 1.5.3**
**Python version 3.5.2 (64-bit)**
**Windows 7**

- Matplotlib installed from Anaconda

**Comment by tacaswell** [comment]
Thanks for tracking down the issue and reporting it!

@settheory Can you open a pull request with that change?

**Comment by anntzer** [comment]
Can you detail how to reproduce the issue?  I cannot reproduce it.

**Comment by tacaswell** [comment]
The line identified in clearly wrong (as `rm` is the wrong type), but I am also not sure how to trigger this.

**Comment by anntzer** [comment]
Closed by #7945
